Cargo.toml

```diff
@@ -64,6 +64,7 @@ ctrlc = "3.1.3"
 prometheus = "0.7.0"
 cargo_metadata = "0.12.1"
 indexmap = "1.4.0"
+git2 = { version = "0.13", features = ["vendored-libgit2"] }
```

Comment:
Can this have a comment on it referencing some kind of upstream bug or similar, that when closed means we can drop this?

Reply:
:facepalm: I misidentified the problem, just like in this comment: https://github.com/rust-lang/git2-rs/issues/746#issuecomment-989711011

Looking into this a bit more, the problem is that distros ship various versions of libgit2, which are not ABI-compatible. As far as I can tell, no matter how you slice it there are going to be build breaks for the forseeable future unless a vendored libgit2 is used.

On `ubuntu:focal` and `ubuntu:latest` I see version 0.28.4 
On `debian:latest` I see version 1.1.0
On Manjaro/Arch I see version 1.3.0

The `Dockerfile` in this repo specifices `ubuntu:focal` and the CI config says `ubuntu-latest` which ship 0.28.4, but the `Cargo.lock` in here says libgit2-sys version 0.12.14+1.1.0. I think at the moment, this repo is just getting lucky that this all manages to work.

Considering that there's apparently breakage between versions 1.1.0 and 1.3.0, I'm not sure if there is any compatibility that can be relied upon between versions of libgit2. Should we have PRs instead to patch `rustwide` and `crates-index`, because `git2` is effectively non-portable without the vendored feature?

Reply:
Someone on the community Discord pointed out to me `libgit2-sys` automatically vendors if the system library is older than the native library it expects, but not if the system library is _newer_. Therefore, this line can be dropped, and I'll ask about the general version incompatibility situation on the `git2` issue about it.